### ISSUE TITLE
security: move ROLE_MAP from env var to database with cross-channel access control

### ISSUE THREAD

**Issue by peter-fusek** [peter-fusek/oncoteam#324]
## Problem
User access rights (phone→patient mappings, roles, permissions) are stored in `NUXT_ROLE_MAP` as a JSON env var on the Railway dashboard service. This caused a P0 outage when Railway lost the env var — ALL WhatsApp users were routed to onboarding.

### Current issues:
1. **Single point of failure**: env var loss = complete WhatsApp lockout
2. **Not shared across channels**: only the dashboard reads NUXT_ROLE_MAP. Backend has separate patient_context.py registry. These can drift.
3. **Privacy concern**: phone numbers + email addresses stored in plain text env var visible in Railway UI
4. **Hard to update**: adding a new user requires Railway env var change + redeploy
5. **No audit trail**: no history of who was granted access when

## Proposed Solution
Store access rights in the database (oncofiles or Postgres) with a secure API:

### Data model
```
access_rights {
  id: int
  email: string (unique)
  phone: string (E.164, unique)  
  name: string
  roles: string[] (advocate, patient, doctor)
  patient_id: string (primary patient)
  patient_ids: string[] (all accessible patients)
  created_at: timestamp
  updated_at: timestamp
  created_by: string
}
```

### Access pattern
- WhatsApp webhook: lookup by phone → get roles + patient_ids
- Dashboard login: lookup by email → get roles + patient_ids
- Backend agents: lookup by patient_id → get all authorized users
- Admin: CRUD via dashboard API (not env var changes)

### Migration path
1. Create DB table / oncofiles collection for access rights
2. Add API endpoints: GET/POST/PUT /api/access-rights
3. Migrate NUXT_ROLE_MAP data to DB
4. Update webhook to read from DB (with cache, fallback to env var)
5. Update dashboard auth to read from DB
6. Remove NUXT_ROLE_MAP env var dependency

### Security requirements
- [ ] Phone numbers encrypted at rest
- [ ] Access changes logged with audit trail
- [ ] Rate-limited access lookup (prevent enumeration)
- [ ] Webhook fails CLOSED if access DB is unreachable (not open to onboarding)
- [ ] Cross-channel consistency: same source for WhatsApp, dashboard, backend agents

## Context
- Sprint 82 incident: Railway lost env vars, WhatsApp down for ~3 hours
- Current workaround: re-adding vars manually as Railway Service Variables
- Related: #323 (WhatsApp down — missing env vars)

**Comment by peter-fusek** [comment]
ROLE_MAP now served from database (commit 6824ab0):

**Backend**:
- `GET /api/internal/access-rights` — reads role_map from oncofiles agent_state (60s cache)
- `POST /api/internal/access-rights` — writes role_map updates

**Dashboard**:
- All 7 consumers migrated from `config.roleMap` (env var) to `getRoleMap()` (DB-backed, 60s TTL)
- Graceful fallback: if backend unreachable, falls back to `NUXT_ROLE_MAP` env var

**Migration path**:
1. Deploy this change ✅
2. Seed current ROLE_MAP data to oncofiles via POST endpoint (next step)
3. NUXT_ROLE_MAP env var becomes fallback cache (keep for now)
4. Future: admin UI for CRUD, then remove env var entirely

**Tests**: 709 passing (4 new access rights tests)
